src/Gateway.php

```diff
@@ -357,6 +357,46 @@ public function init_form_fields() {
 			],
 		];
 
+		if ( \in_array( $this->payment_method, [ PaymentMethods::IDEAL, PaymentMethods::DIRECT_DEBIT_IDEAL ], true ) ) {
+			$default = 'no';
+
+			$gateway_post_date = null;
+
+			$gateway = Plugin::get_gateway( (int) $this->config_id );
+
+			if ( null !== $gateway ) {
+				$payment_method_object = $gateway->get_payment_method( $this->payment_method );
+
+				if ( null !== $payment_method_object ) {
+					foreach ( $payment_method_object->get_fields() as $field ) {
+						if ( 'issuer' !== $field->meta_key ) {
+							continue;
+						}
+
+						$default_disabled_start_date = new DateTime( '2025-01-01' );
+
+						$gateway_post_date = \get_post_datetime( (int) $this->config_id );
+
+						if ( $gateway_post_date < $default_disabled_start_date ) {
+							$default = 'yes';
+						}
+
+						break;
+					}
+				}
+			}
+
+			if ( null === $gateway || null !== $gateway_post_date ) {
+				$this->form_fields['show_ideal_issuers'] = [
+					'type'        => 'checkbox',
+					'title'       => \__( 'Show iDEAL issuers', 'pronamic_ideal' ),
+					'label'       => \__( 'Show legacy iDEAL issuer selection during checkout', 'pronamic_ideal' ),
+					'description' => \__( 'Support for this feature will be removed as issuer selection has moved from checkout to the iDEAL 2.0 payment screen. It is recommended to disable the legacy iDEAL issuer selection.', 'pronamic_ideal' ),
+					'default'     => $default,
+				];
+			}
+		}
+
```

Comment:
Would it also be more clear to move this to its own private method, so that the `init_form_fields()` function does not increase much in complexity? This also gives the opportunity to make more use of early returns, which for me improves readability.

Does it have any added value to check the iDEAL payment method fields for an issuer field? I would simplify it to 'Show iDEAL issuers if available' or something like that.

> Support for this feature will be removed as issuer selection has moved from checkout to the iDEAL 2.0 payment screen.

I would phrase this a little differently because it's not clear that this functionality will actually be removed. Maybe something more neutral like:

> With the launch of the new iDEAL (2.0) in mid-2024, it is advised to let customers make their bank choice on the new iDEAL payment screen. It is therefore not recommended to show the iDEAL banks on your own website.

Reply:
> Does it have any added value to check the iDEAL payment method fields for an issuer field? I would simplify it to 'Show iDEAL issuers if available' or something like that.

A user can then enable the 'Show iDEAL issuers' setting and reference to it in the description, while the gateway payment method does not support issuers? I would try to prevent that. Additionally, when we remove the iDEAL issuers field from gateways in the future (according to https://help.mollie.com/hc/nl/articles/19100313768338-iDEAL-2-0 for example, all users should migrate to iDEAL 2.0 before March 1, 2025 — which is done by removing the issuer selection from checkout, so at some point I imagine we remove the issuers field from the iDEAL payment method in the Mollie gateway), this temporary setting will also be phased out automatically and it will be clear if it can be used instead of only mentioning "if available" and let users figure out if it can be used. Without this test, the setting will even be available for gateways which have never supported iDEAL issuers, which seems kind of strange to me.

Reply:
> I would phrase this a little differently because it's not clear that this functionality will actually be removed.

As far as I'm aware, it's quite clear that the possibility of issuer selection in the shop is temporary and will be removed. The iDEAL implementation docs at https://currencenl.atlassian.net/wiki/spaces/IPD/pages/3432972289/Generic+Merchant+info+and+requirements+on+the+New+iDEAL#Issuer-Selection-and-IssuerID for example mention:

> - The Merchant MUST NOT provide an Issuer list to the User together with his iDEAL payment button. Issuer selection will be done on the iDEAL Payment Page.
>   - Only for a limited period of time, Merchants that have not fully implemented the new iDEAL MAY still provide an Issuer list to the User.

Reply:
> Would it also be more clear to move this to its own private method, so that the init_form_fields() function does not increase much in complexity? This also gives the opportunity to make more use of early returns, which for me improves readability.

I assume that would be possible and I'll look into that 👀

Reply:
> Would it also be more clear to move this to its own private method […]

Moved to private method in https://github.com/pronamic/wp-pronamic-pay-woocommerce/pull/74/commits/ca28e3bcf0f510c3f8847c6fe0f1a36d3f2cd5d9.